File: algo/contains-duplicate-ii/containsNearbyDuplicate.go
func containsNearbyDuplicate(nums []int, k int) bool {
    hm := map[int]int{}
    for i, n := range nums {
        if e, ok := hm[n]; ok {
            if i - e > k {
                hm[n] = i
            } else {
                return true
            }
        } else {
            hm[n] = i
        }
    }
    return false
}

File: algo/contains-duplicate-ii/containsNearbyDuplicate.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        hm = {}
        for i, n in enumerate(nums):
            if n in hm:
                if i - hm[n] > k:
                    hm[n] = i
                else:
                    return True
            else:
                hm[n] = i
        return False
